components/raftstore-v2/src/operation/ready/apply_trace.rs

```diff
@@ -519,6 +522,18 @@ impl<EK: KvEngine, ER: RaftEngine> Storage<EK, ER> {
         }
         let region_id = self.region().get_id();
         let raft_engine = self.entry_storage().raft_engine();
+        let last_epoch = self.region().get_region_epoch();
+        if util::is_epoch_stale(self.flushed_epoch(), last_epoch) {
```

Comment:
why not compare self.flushed_epoch with epoch of line527 directly?

Reply:
I afraid  the cost of `raft_engine.get_region_state()` is high.

Reply:
I think it's fine. `get_region_state` is an in-memory operation and furthermore, `record_apply_trace` is rarely called.